File: 917 - Reverse Only Letters.java
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Stack;

class Solution {
    public String reverseOnlyLetters(String S) {
        Deque<Character> s = new ArrayDeque<>();
        for (Character c : S.toCharArray()) {
            if (Character.isAlphabetic(c)) {
                s.push(c);
            }
        }
        String res = "";
        for (Character c : S.toCharArray()) {
            if (Character.isAlphabetic(c)) {
                res += s.pop();
            } else {
                res += c;
            }
        }
        return res;
    }
}

File: 917 - Reverse Only Letters.js
/**
 * @param {string} S
 * @return {string}
 */
var reverseOnlyLetters = function (S) {
    let stack = [];
    let patt = /[a-zA-Z]/;
    for (let c of S) {
        if (patt.test(c)) {
            stack.push(c);
        }
    }
    let res = '';
    for (let c of S) {
        if (patt.test(c)) {
            res += stack.pop();
        } else {
            res += c;
        }
    }
    return res;

};